src/Arduino_GigaDisplayTouch.h

```diff
@@ -77,19 +172,19 @@ class Arduino_GigaDisplayTouch {
      * @param addr The device address for the touch controller.
      */
     #if defined(ARDUINO_GIGA)
-      Arduino_GigaDisplayTouch(TwoWire& wire  = Wire1, 
+      Arduino_GigaDisplayTouch(TwoWire& wire  = Wire1,
                                uint8_t intPin = PinNameToIndex(PI_1),
-                               uint8_t rstPin = PinNameToIndex(PI_2), 
+                               uint8_t rstPin = PinNameToIndex(PI_2),
                                uint8_t addr   = GT911_I2C_ADDR_BA_BB);
     #elif defined(ARDUINO_PORTENTA_H7_M7)
-      Arduino_GigaDisplayTouch(TwoWire& wire  = Wire, 
+      Arduino_GigaDisplayTouch(TwoWire& wire  = Wire,
                                uint8_t intPin = PinNameToIndex(PD_4),
-                               uint8_t rstPin = PinNameToIndex(PD_5), 
+                               uint8_t rstPin = PinNameToIndex(PD_5),
                                uint8_t addr   = GT911_I2C_ADDR_BA_BB);
-    #else 
-      Arduino_GigaDisplayTouch(TwoWire& wire, 
-                               uint8_t intPin, 
-                               uint8_t rstPin, 
+    #else
```

Comment:
Do we need to duplicate the class?
```suggestion
    #elif defined(__ZEPHYR__)
    Arduino_GigaDisplayTouch();
    #else
```

Reply:
Note: I pushed up a clang-format of the files and then restricted these changes to the zephyr area.

This is sort of a global question, that we (@mjs513 and I) have been trying to ask. How does Arduino want to handle some of these libraries?  There are several different options:

a) Completely different libraries, that potentially part of the Board installs.  Like: Camera, Wire, SPI are done now

b) As libraries you install from the library manager.  With this there are at least 3 variations that I have seen (and done)
1) Try to intersperse all the code the MBED and Zephyr stuff with lots of #ifdef...
2) Like I did, where there is like 1 #ifdef or the like with the two different implementations.
3) like 2, except that split the files up. like you might have Arduio_GigaDisplayTouch_zephyr.cpp and these have
the same #if stuff like 2, but easier to locate the code and defines.

There are pros and cons to all of these.

So again not sure which way?

Reply:
I don't think there's a single rule that fits all, this is probably best decided on a case-by-case basis, so please use your better judgement. For example, if you find yourself adding a whole new alternate implementation, then it should probably go in a new file. However, there might be an argument for keeping everything in the same file if support for Mbed and other cores is planned to be removed, that would make it easier to do so. I'm also not sure if this is planned or not, perhaps @pennam knows more about this. For now, the header file definitely works with interspersed `ifdef`'s.  

> Completely different libraries, that potentially part of the Board installs. Like: Camera, Wire, SPI are done now

I think core libraries provide core functionality like Wire and SPI. Libraries like touchscreen, BLE etc.. are not core functionality.